cli/internal/mock/plugin.go

```diff
@@ -23,24 +23,49 @@ import (
 
 // Mock plugin used only for testing.
 type mockPlugin struct {
-	fail bool
+	fail         bool
+	envGenerator bool
 }
 
 func NewPlugin(failPlugin bool) *mockPlugin {
-	return &mockPlugin{fail: failPlugin}
+	return &mockPlugin{
+		fail:         failPlugin,
+		envGenerator: true,
+	}
+}
+
+func NewSigGeneratorPlugin(failPlugin bool) *mockPlugin {
+	return &mockPlugin{
+		fail:         failPlugin,
+		envGenerator: false,
+	}
 }
 
 func (p *mockPlugin) DescribeKey(ctx context.Context, req *plugin.DescribeKeyRequest) (*plugin.DescribeKeyResponse, error) {
-	return nil, plugin.NewUnsupportedError("DescribeKey operation is not implemented by example plugin")
+	if p.fail || p.envGenerator {
+		return nil, fmt.Errorf("DescribeKey() expected error")
+	}
+	return &plugin.DescribeKeyResponse{
+		KeyID:   "someKeyId",
+		KeySpec: plugin.KeySpecRSA2048,
+	}, nil
 }
 
 func (p *mockPlugin) GenerateSignature(ctx context.Context, req *plugin.GenerateSignatureRequest) (*plugin.GenerateSignatureResponse, error) {
-	return nil, plugin.NewUnsupportedError("GenerateSignature operation is not implemented by example plugin")
+	if p.fail || p.envGenerator {
+		return nil, fmt.Errorf("GenerateSignature() expected error")
+	}
+	return &plugin.GenerateSignatureResponse{
+		KeyID:            "someKeyId",
+		Signature:        []byte("abcd"),
+		SigningAlgorithm: plugin.SignatureAlgorithmRSASSA_PSS_SHA256,
+		CertificateChain: [][]byte{[]byte("abcd"), []byte("wxyz")},
+	}, nil
 }
 
 func (p *mockPlugin) GenerateEnvelope(ctx context.Context, req *plugin.GenerateEnvelopeRequest) (*plugin.GenerateEnvelopeResponse, error) {
-	if p.fail {
```

Comment:
nit:
on line 72 `SignatureEnvelopeType: "application/jose+json",`, this should be
`SignatureEnvelopeType: req.SignatureEnvelopeType,`